### PR TITLE
[SPARK-46610][SQL] Create table should throw exception when no value for a key in options

### PR DESCRIPTION
**Pull request by amaliujia**
### What changes were proposed in this pull request?

Before SPARK-43529, there was a check from `visitPropertyKeyValues` that throws for null values for option keys. After SPARK-43529, a new function is used to support expressions in options but the new function lose the check.

This PR adds the check back.

### Why are the changes needed?

Throw exception when a option value is null.

### Does this PR introduce _any_ user-facing change?

NO

### How was this patch tested?

UT

### Was this patch authored or co-authored using generative AI tooling?

NO

### CODE REVIEW

**Comment by amaliujia** [comment]
@cloud-fan

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala [hunk 1]
```diff
@@ -3380,6 +3380,10 @@ class AstBuilder extends DataTypeAstBuilder with SQLConfHelper with Logging {
     val options = ctx.expressionProperty.asScala.map { property =>
       val key: String = visitPropertyKey(property.key)
       val value: Expression = Option(property.value).map(expression).orNull
+      if (value == null) {
+        operationNotAllowed(
+          s"Value must be specified for key: $key.", ctx)
+      }
```
Comment (MaxGekk) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala:3386]:
```suggestion
      val value: Expression = Option(property.value).map(expression).getOrElse {
        operationNotAllowed(s"A value must be specified for the key: $key.", ctx)
      }
```

**Review by cloud-fan — APPROVED** [review]

**Comment by amaliujia** [comment]
@cloud-fan tests all green

**Comment by cloud-fan** [comment]
thanks, merging to master/3.5!

**Pull request closed without merging**